engine/api/application.go

```diff
@@ -439,34 +451,38 @@ func (api *API) updateApplicationHandler() service.Handler {
 
 		tx, err := api.mustDB().Begin()
 		if err != nil {
-			return sdk.WrapError(err, "Cannot start transaction")
+			return sdk.WrapError(err, "cannot start transaction")
 		}
 		defer tx.Rollback() // nolint
-		if err := application.Update(tx, app); err != nil {
-			return sdk.WrapError(err, "Cannot delete application %s", applicationName)
+
+		if err := application.Update(ctx, tx, app); err != nil {
+			return sdk.WrapError(err, "cannot delete application %s", applicationName)
 		}
 
 		if err := tx.Commit(); err != nil {
 			return sdk.WithStack(err)
 		}
 
-		event.PublishUpdateApplication(ctx, p.Key, *app, old, getAPIConsumer(ctx))
+		event.PublishUpdateApplication(ctx, proj.Key, *app, old, getAPIConsumer(ctx))
 
 		return service.WriteJSON(w, app, http.StatusOK)
-
 	}
 }
 
 func (api *API) postApplicationMetadataHandler() service.Handler {
 	return func(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
-		// Get pipeline and action name in URL
 		vars := mux.Vars(r)
 		projectKey := vars[permProjectKey]
 		applicationName := vars["applicationName"]
 
-		app, err := application.LoadByName(api.mustDB(), projectKey, applicationName)
+		proj, err := project.Load(api.mustDB(), projectKey)
```

Comment:
I think it's a bit sad to load the project to get its ID, whereas the projectKey is an identifier.
we should consider `application.LoadByProjectKeyAndName`